### PR TITLE
Catch parser's exception when azure-webapp:config

### PR DESCRIPTION
**Pull request by Flanker32**
Separate the parser and validator, and show warnings while `azure-webapp:config`.
image

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/configuration/RuntimeSetting.java [hunk 1]
```diff
@@ -40,20 +40,20 @@ public OperatingSystemEnum getOsEnum() {
...
     public JavaVersion getJavaVersion() {
-        return StringUtils.isEmpty(javaVersion) ? null : JavaVersion.fromString(javaVersion);
+        return (StringUtils.isEmpty(javaVersion) || !checkJavaVersion(javaVersion)) ?
+                null : JavaVersion.fromString(javaVersion);
     }
 
-    public RuntimeStack getLinuxRuntime() throws MojoExecutionException {
+    public RuntimeStack getLinuxRuntime() {
         // todo: add unit tests
         final RuntimeStack result = RuntimeStackUtils.getRuntimeStack(javaVersion, webContainer);
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/configuration/RuntimeSetting.java]: return RuntimeStackUtils.getRuntimeStack(javaVersion, webContainer);

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/ConfigurationValidator.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.maven.webapp.validator;
+
+import com.microsoft.azure.maven.webapp.AbstractWebAppMojo;
+import org.codehaus.plexus.util.StringUtils;
+
+public abstract class ConfigurationValidator {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/ConfigurationValidator.java]: AbstractConfigurationValidator

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/V2ConfigurationValidator.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if (runtime == null) {
+            return "Pleas config the <runtime> in pom.xml.";
+        }
+        if (runtime.getWebContainer() == null) {
+            return "The configuration <webContainer> in pom.xml is not correct.";
+        }
+        return null;
+    }
+
+    public String validateImage() {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/V2ConfigurationValidator.java]: two validateImage with different logic?

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/V2ConfigurationValidator.java]: Yes, for V1 schema the parent node of `<image>` is `<containerSettings>` while V2 is `<runtime>`

**Pull request merged**